crates/tap-agent/tests/sender_account_test.rs

```diff
@@ -0,0 +1,102 @@
+// Copyright 2023-, Edge & Node, GraphOps, and Semiotic Labs.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{collections::HashSet, str::FromStr};
+
+use indexer_tap_agent::{
+    agent::sender_account::SenderAccountMessage,
+    test::{create_received_receipt, create_sender_account, get_grpc_url, store_receipt},
+};
+use ractor::concurrency::Duration;
+use reqwest::Url;
+use serde_json::json;
+use sqlx::PgPool;
+use test_assets::{ALLOCATION_ID_0, TAP_SIGNER as SIGNER};
+use thegraph_core::alloy::hex::ToHexExt;
+use wiremock::{
+    matchers::{body_string_contains, method},
+    Mock, MockServer, ResponseTemplate,
+};
+
+const TRIGGER_VALUE: u128 = 500;
+const RECEIPT_LIMIT: u64 = 10000;
+
+// This test should ensure the full flow starting from
+// sender account layer to work, up to closing an allocation
+#[sqlx::test(migrations = "../../migrations")]
+async fn sender_account_layer_test(pgpool: PgPool) {
+    let mock_server = MockServer::start().await;
+    let mock_escrow_subgraph_server: MockServer = MockServer::start().await;
+    mock_escrow_subgraph_server
+        .register(Mock::given(method("POST")).respond_with(
+            ResponseTemplate::new(200).set_body_json(json!({ "data": {
+                "transactions": [],
+            }
+            })),
+        ))
+        .await;
+
+    let receipt = create_received_receipt(&ALLOCATION_ID_0, &SIGNER.0, 1, 1, TRIGGER_VALUE - 100);
+    store_receipt(&pgpool, receipt.signed_receipt())
+        .await
+        .unwrap();
+
+    let (sender_account, notify, _, _) = create_sender_account()
+        .pgpool(pgpool.clone())
+        .rav_request_trigger_value(TRIGGER_VALUE)
+        .max_amount_willing_to_lose_grt(TRIGGER_VALUE + 1000)
+        .escrow_subgraph_endpoint(&mock_escrow_subgraph_server.uri())
+        .network_subgraph_endpoint(&mock_server.uri())
+        .rav_request_receipt_limit(RECEIPT_LIMIT)
+        .aggregator_endpoint(Url::from_str(&get_grpc_url().await).unwrap())
```

Comment:
Not needed:
- aggregator_endpoint
- rav_request_receipt_limit
- rav_request_trigger_value.